index.js

```diff
@@ -71,7 +71,11 @@ module.exports = function staticCache(dir, options, files) {
 
       var s
       try {
-        s = yield fs.stat(path.join(dir, filename))
+        var fullpath = path.join(dir, filename)
+        if (fullpath.indexOf(dir) === -1) {
```

Comment:
should normalize dir in the beginning too.

`fullpath.indexOf(dir) !== 0` will be better.